src/components/NotificationDot/NotificationDot.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { cloneElement, Children, ReactElement } from "react";
+import styled from "styled-components";
+import { NotificationDotProps, DotProps } from "./types";
+
+const NotificationDotRoot = styled.span`
+  display: inline-flex;
+  position: relative;
+`;
+
+const Dot = styled.span<DotProps>`
+  display: ${({ invisible }) => (invisible ? "none" : "inline-flex")};
+  position: absolute;
+  top: 0;
+  right: 0;
+  width: 10px;
+  height: 10px;
+  pointer-events: none;
+  border: 1px solid ${({ theme }) => theme.colors.invertedContrast};
+  border-radius: 50%;
+  background-color: ${({ theme }) => theme.colors.failure};
+`;
+
+const NotificationDot: React.FC<NotificationDotProps> = ({ invisible = false, children, ...props }) => (
```

Comment:
Can we flip the `invisible` prop? Instead of negative assertion can we use a positive one. By default it should _not_ show so the prop name should be something like `show`.